### PR TITLE
[docs] Provide recipe for running Chrome and Firefox with Travis

### PR DESCRIPTION
**Pull request by AndreyBelym**
Running example: https://travis-ci.org/AndreyBelym/testcafe-example-travis-local/builds/184244319

Requires https://github.com/DevExpress/testcafe/pull/1068

\cc @DevExpress/testcafe-docs

### CODE REVIEW

File: docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+```
+
+For more information on how to configure a test run using a `testcafe` command, see [Command Line Interface](../using-testcafe/command-line-interface.md).
+
+**Important:** If you are going to run tests for a website that is not deployed, the `test` script should also include commands to run and stop the site; for example:
+
+```text
+"scripts": {
+  "test":  "node server.js & testcafe chrome,firefox tests/index-test.js; EXIT_CODE=$?; kill $!; exit $EXIT_CODE"
```
Comment (inikulin) [inline docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md]: I suggest to use `concurrently` package, as shown here: http://devexpress.github.io/testcafe/documentation/recipes/test-static-html-pages.html

Note you also need to specify `-s first` flag to concurrently, it's missing in original topic.

**Comment by AndreyBelym** [comment]
FPR

File: docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+For more information on how to configure a test run using a `testcafe` command, see [Command Line Interface](../using-testcafe/command-line-interface.md).
+
+**Important:** If you are going to run tests for a website that is not deployed, the `test` script should also include commands to run and stop the site. 
+For example, install [concurrently](https://github.com/kimmobrunfeldt/concurrently) and add `following to your package.json`: 
+
+```text
+"scripts": {
+  "test":  "concurrently -s last -r -k 'node' 'node server.js' 'testcafe chrome,firefox tests/index-test.js'"
```
Comment (miherlosev) [inline docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md]: Option `-k` doesn't have parameters

**Comment by AndreyBelym** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
@AndreyBelym I'll take care about this PR. Enjoy your vacation =)

File: docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+     Commit and push this file to your repository.
+
+## Step 3 - Configure Travis to use Xvfb
+
+Travis CI uses Ubuntu Server virtual machines, that do not have regular graphical environment like GNOME or KDE installed, So you have to setup and use [Xvfb](https://www.x.org/archive/X11R7.6/doc/man/man1/Xvfb.1.xhtml) to run browsers headlessly. 
+
+You should add the following sections to your `.travis.yml` to start `Xvfb`:
+
+```
```
Comment (inikulin) [inline docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md]: Highlight as `yaml`

File: docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+     node_js: "stable"
+     ```
+
+     Commit and push this file to your repository.
+
+## Step 3 - Configure Travis to use Xvfb
+
+Travis CI uses Ubuntu Server virtual machines, that do not have regular graphical environment like GNOME or KDE installed, So you have to setup and use [Xvfb](https://www.x.org/archive/X11R7.6/doc/man/man1/Xvfb.1.xhtml) to run browsers headlessly. 
+
+You should add the following sections to your `.travis.yml` to start `Xvfb`:
```
Comment (inikulin) [inline docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md]: Never use `you should` in docs

File: docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+```text
+"scripts": {
+    "test":  "testcafe chrome,firefox tests/index-test.js"
+}
+```
+
+For more information on how to configure a test run using a `testcafe` command, see [Command Line Interface](../using-testcafe/command-line-interface.md).
+
+**Important:** If you are going to run tests for a website that is not deployed, the `test` script should also include commands to run the site.
```
Comment (inikulin) [inline docs/articles/documentation/recipes/running-tests-in-firefox-and-chrome-using-travis-ci.md]: I don't like it in the form of `if you need, then you should`. It should smthg like: "if you need to run tested app server as well, you can use TestCafe `--app-command` flag: TestCafe will start server before testing and terminate it once the testing is done". Also, it shouldn't be marked as `Important`. It's just a note

File: docs/articles/documentation/recipes/test-static-html-pages.md [hunk 6]
```diff
@@ -8,33 +8,33 @@ permalink: /documentation/recipes/test-static-html-pages.html
...
+    "test": "testcafe chrome ./test/acceptance/** --app 'http-server ./dist -s'"
 }
 ```
 
 This script contains the following commands.
 
-1. `concurrently -r -k "http-server"` - starts concurrently with the raw output flag on and kills the http-server after running the tests
-2. `"http-server ./dist -p 4000 -s"` - starts the local server at port `4000` on the `./dist folder` in silent mode
-3. `"testcafe chrome ./test/acceptance/**"` - runs TestCafe tests on the `./test/acceptance` folder on Chrome
+1. `"http-server ./dist -s"` - starts the local server at port `8080` with files from the `./dist` folder in silent mode
```
Comment (inikulin) [inline docs/articles/documentation/recipes/test-static-html-pages.md]: BTW, we need to specify here what should be set as `fixture.page` here

File: docs/articles/documentation/recipes/running-tests-using-travis-ci-and-sauce-labs.md [hunk 7]
```diff
@@ -77,22 +77,16 @@ To tell npm how to run your tests, you need to add the `test` script to the proj
...
 
-**Important:** If you are going to run tests for a website that is not deployed, the `test` script should also include commands to run and stop the site; for example:
+**Important:** If you are going to run tests for a website that is not deployed, the `test` script should also include commands to run the site. 
+Use the `--app` TestCafe option to specify a command that deploys the website locally.
+This command will be automatically executed before running tests. After tests are finished, TestCafe will stop the server.
 
 ```text
 "scripts": {
-  "test":  "node server.js & testcafe 'saucelabs:Chrome@beta:Windows 10' tests/index-test.js; EXIT_CODE=$?; kill $!; exit $EXIT_CODE"
+  "test":  "testcafe saucelabs:Chrome@beta tests/index-test.js --app 'node server.js'"
```
Comment (inikulin) [inline docs/articles/documentation/recipes/running-tests-using-travis-ci-and-sauce-labs.md]: Here and in other places don't use `'` in npm commands, it's interpreted incorrectly (at least on Windows), use escaped `\"` instead:
```json
"test":  "testcafe saucelabs:Chrome@beta tests/index-test.js --app \"node server.js\""
```

File: examples/running-tests-using-travis-and-saucelabs/package.json [hunk 8]
```diff
@@ -4,7 +4,7 @@
   "description": "An example of how to run TestCafe tests in the cloud using Travis CI and Sauce Labs.",
   "main": "server.js",
   "scripts": {
-    "test": "node server.js & testcafe 'saucelabs:Chrome@beta:Windows 10' tests/index-test.js; EXIT_CODE=$?; kill $!; exit $EXIT_CODE"
+    "test": "testcafe 'saucelabs:Chrome@beta:Windows 10' tests/index-test.js --app 'node server.js'"
```
Comment (inikulin) [inline examples/running-tests-using-travis-and-saucelabs/package.json]: Use escaped `\"` here

File: examples/running-tests-in-firefox-and-chrome-using-travis-ci/package.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "running-tests-using-travis-and-saucelabs",
+  "version": "1.0.0",
+  "description": "An example of how to run TestCafe tests in the cloud using Firefox and Chrome with Travis CI.",
+  "main": "server.js",
+  "scripts": {
+    "test": "testcafe chrome,firefox tests/index-test.js --app 'node server.js'"
```
Comment (inikulin) [inline examples/running-tests-in-firefox-and-chrome-using-travis-ci/package.json]: Use escaped `\"` here

**Comment by AndreyBelym** [comment]
FPR

File: examples/running-tests-using-travis-and-saucelabs/package.json [hunk 10]
```diff
@@ -23,6 +23,7 @@
   "devDependencies": {
     "testcafe": "*",
     "chai": "^3.5.0",
+	"concurrently": "^3.1.0",
```
Comment (miherlosev) [inline examples/running-tests-using-travis-and-saucelabs/package.json]: Is it the unused dependency?

File: examples/running-tests-in-firefox-and-chrome-using-travis-ci/package.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    "url": "https://github.com/DevExpress/testcafe/issues"
+  },
+  "homepage": "https://github.com/DevExpress/testcafe#readme",
+  "dependencies": {
+    "connect": "^3.4.1",
+    "serve-static": "^1.11.1"
+  },
+  "devDependencies": {
+    "testcafe": "*",
+    "chai": "^3.5.0"
```
Comment (miherlosev) [inline examples/running-tests-in-firefox-and-chrome-using-travis-ci/package.json]: Unused dependency

**Pull request merged**